scipy/signal/signaltools.py

```diff
@@ -873,10 +873,52 @@ def lfilter(b, a, x, axis=-1, zi=None):
     """
     if isscalar(a):
         a = [a]
-    if zi is None:
-        return sigtools._linear_filter(b, a, x, axis)
+    if a[0] == 1 and len(a) == 1:
```

Comment:
The docstring promises that if ``a[0]`` is not 1, then ``a`` and ``b`` are normalized by ``a[0]``. So probably this path should be used for all length-1 ``a``?